code/datums/status_effects/xeno_buffs.dm

```diff
@@ -823,3 +823,89 @@
 	scale = 0.6
 	rotation = 0
 	spin = 0
+
+// ***************************************
+// *********** Blessings
+// ***************************************
+/datum/status_effect/blessing_frenzy
+	id = "blessing of frenzy"
+	duration = 25 SECONDS
+	tick_interval = -1
+	status_type = STATUS_EFFECT_REFRESH
+	alert_type = null
+	/// The owner of this buff.
+	var/mob/living/carbon/xenomorph/buff_owner
+	///Aura strength of the puppeteer who gave this effect
+	var/strength = 1
+
+/datum/status_effect/blessing_frenzy/on_creation(mob/living/new_owner, aura_strength)
+	owner = new_owner
+	strength = aura_strength
+	return ..()
+
+/datum/status_effect/blessing_frenzy/on_apply()
+	buff_owner = owner
+	if(!isxeno(buff_owner))
+		return FALSE
+	buff_owner.add_movespeed_modifier(type, TRUE, 0, NONE, TRUE, strength * -0.2)
+	return TRUE
+
+/datum/status_effect/blessing_frenzy/on_remove()
+	buff_owner.remove_movespeed_modifier(type)
+	return ..()
+
+/datum/status_effect/blessing_fury
+	id = "blessing of fury"
+	duration = 25 SECONDS
+	tick_interval = -1
+	status_type = STATUS_EFFECT_REFRESH
+	alert_type = null
+	/// The owner of this buff.
+	var/mob/living/carbon/xenomorph/buff_owner
+	var/initial_modifier
```

Comment:
You should save the change instead and +- that since xeno_melee_damage_modifier can be modified by other things

ps autodoc